### PR TITLE
Make ConfigSSL (..) public

### PR DESCRIPTION
**Pull request by erikd**
I have tested that this compiles in my project, but not yet tested it (because I still do not know how).

### CODE REVIEW

File: src/Foreign/JavaScript.hs [hunk 1]
```diff
@@ -18,7 +18,7 @@ module Foreign.JavaScript (
         , jsCustomHTML, jsStatic, jsLog
         , jsWindowReloadOnDisconnect, jsCallBufferMode
         , jsUseSSL),
-    ConfigSSL (jsSSLBind, jsSSLCert, jsSSLChainCert, jsSSLKey, jsSSLPort),
+    ConfigSSL (..),
```
Comment (HeinrichApfelmus) [inline src/Foreign/JavaScript.hs:21]: Ah. The wildcard also exports the constructor — which I do not want to export in order to preserve backwards compatibility.

File: src/Graphics/UI/Threepenny/Core.hs [hunk 2]
```diff
@@ -5,7 +5,7 @@ module Graphics.UI.Threepenny.Core (
 
     -- * Server
     -- $server
-    Config(..), defaultConfig, startGUI,
+    Config(..), ConfigSSL (..), defaultConfig, startGUI,
```
Comment (HeinrichApfelmus) [inline src/Graphics/UI/Threepenny/Core.hs:8]: 👍

**Comment by erikd** [comment]
So you are ok with the `ConfigSSL` constructor being public, but not the `Config` one?

File: src/Foreign/JavaScript.hs (hunk 1 above)
Comment (erikd) [reply src/Foreign/JavaScript.hs:21]: The constructor is needed in order to actually _construct_ an `ConfigSSL` value. I also do not think its possible to provide a sensible default for those fields.

**Pull request merged**